docs/concepts/results.md

```diff
@@ -290,9 +290,13 @@ Persistence of results requires a [**serializer**](#result-serializers) and a [*
 - `result_storage`: Where to store the result when persisted.
 - `result_serializer`: How to convert the result to a storable form.
 
+The flow decorator also includes the option:
+
+- `persist_results`: Whether the result of this flow and any tasks or subflows in it should be persisted to storage.
```

Comment:
😬 Are we sure it's a good idea to have two options that differ only by `s`?  I feel like it will be super easy to mix that up, and hard to verbally talk about with peers.  Could we consider renaming this to something like `persist_all_results`?

Reply:
Yea, I agree with @chrisguidry that this will inevitably confuse folks - is there a production use case for disabling all result persistence, or is it just for testing?

Reply:
I was also concerned and a few votes that is confusing is enough for sure. I do not think this should proceed with this variable name. I kind of like `persist_all_results`.

For other's watching, we are chatting about the use-case internally at https://prefecthq.slack.com/archives/C01QFU3549W/p1665775026247199

Reply:
@cicdw to give context: some users want to globally disable result persistence for a flow because e.g. they operate on large dataframes with multiple tasks and they may not even need retries/caching, so this feature is very useful and something we've been asked about in the community

Reply:
Persistence is off by default in 2.6+ — unless you have enabled retries or caching, there is no persistence unless you explicitly opt int.

I see the use-case here as more focused on _enabling_ persistence for all tasks without turning on a feature.

Reply:
I wish there was a way to say `persist_results_by_default` with fewer words.

Boolean options are clean and simple to explain, but there's another possibility here: 
 deprecate `persist_result=bool` and use an enum to get very explicit:

```
persist_results=Persist.Mine
persist_results=Persist.MineAndMySubTasks
persist_results=Persist.OnlyMySubtasks
```

These names are bad, but you see where I'm going...

`persist_result=bool` would translate internally to `persist_results=Persist.Mine` (I think that's the current behavior)

Naturally, this is a lot more complicated, but it gives us a way to avoid ambiguity

Reply:
Whilst I appreciate the clarity of enums, I don't think they're particularly nice for user discoverability and I think they're useful when there are ~3 cases at most. Here we'd have to cover disabling options and clarify which options affect subflows as well.

> I wish there was a way to say persist_results_by_default with fewer words.

I actually don't think that's a bad suggestion. It's clearer than anything else we've discussed here :)

Reply:
I think it makes sense to err on the side of explicitness with `persist_results_by_default`

Reply:
+1, LGTM :)

Reply:
After further thought, this may unfortunately not the clearest option. If `persist_results_by_default` is `False`, we will not persist results when a feature is used that would normally toggle persistence on. In this case, it is not changing the "default" but rather forcing all results to be disabled. I can't think of many use-cases where setting it to `False` makes sense, maybe we'd just want to throw an error if you were to do so when using features that typically require persistence. I'm not sure yet.